app/controllers/redmine_oauth_controller.rb

```diff
@@ -29,11 +36,12 @@ def oauth_github_callback
       unverified = emails.select { |e| !e['verified'] }.map { |v| v['email'] }
       primary = emails.select { |e| e['primary'] }.first.try(:[], 'email')
 
-      if (user = User.where(:mail => verified).first)
-        checked_try_to_login user.mail, user_info
+      if (user = User.find_by_mail(verified))
```

Comment:
this is causing an issue when user have more than one verified email, verified can be array